.github/workflows/ci-sanitizers.yml

```diff
@@ -0,0 +1,128 @@
+name: CI with sanitizers enabled
+
+on:
+  # TODO: Enable pull request and cron triggers later, when this workflow is tested and not failing
+  # pull_request:
+  # schedule:
+    # - cron: '0 20 * * *' # Every day at 12pm PST (UTC-8)
+  workflow_dispatch:
+
+env:
+  BUILD_TYPE: Debug
+  MOUNT: /azure-osconfig
+  REGISTRY: ghcr.io
+  SANITIZER_FLAGS: -fsanitize=address,undefined -fno-omit-frame-pointer -g -O1
+
+jobs:
+  create-ci-matrix:
+    name: Build CI distro matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.get-matrix.outputs.matrix }}
+    steps:
+      - name: Get matrix
+        id: get-matrix
+        # Skip centos-7, oraclelinux-7, rhel-7 as they don't currently build properly from Dockerfile
+        run: |
+          matrix="$(cat <<'EOL'
+          [
+            { "name": "ubuntu-22.04", "arch": "amd64", "tag": "latest" }
+          ]
+          EOL
+          )"
+
+          echo Distros to perform CI on: $matrix
+          echo matrix=$matrix >> $GITHUB_OUTPUT
+
+  unit-test:
+    name: Unit test
+    needs: create-ci-matrix
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        target: ${{ fromJson(needs.create-ci-matrix.outputs.matrix) }}
+          
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          clean: true
+
+      - name: Docker login
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ github.token }}
+
+      # - name: Setup QEMU
+      #   if: ${{ matrix.target.arch != 'amd64' }}
+      #   uses: docker/setup-qemu-action@v3
+
+      - name: Run container
+        id: container
+        uses: ./.github/actions/container-run
+        with:
+          registry: ${{ env.REGISTRY }}
+          container: azure/azure-osconfig/${{ matrix.target.name }}-${{ matrix.target.arch }}
+          mount: ${{ github.workspace }}:${{ env.MOUNT }}
+          tag: ${{ matrix.target.tag }}
+
+      - name: Generate build
+        uses: ./.github/actions/container-exec
+        with:
+          container: ${{ steps.container.outputs.id }}
+          cmd: |
+            mkdir build && cd build
+            cmake ../src -DCMAKE_C_FLAGS="${{ env.SANITIZER_FLAGS }}" -DCMAKE_CXX_FLAGS="${{ env.SANITIZER_FLAGS }}" -DCMAKE_build-type=${{ env.BUILD_TYPE }} -Duse_prov_client=ON -Dhsm_type_symm_key=ON -DCOMPILE_WITH_STRICTNESS=ON -DBUILD_TESTS=ON -DBUILD_SAMPLES=ON -DBUILD_ADAPTERS=ON -Duse_default_uuid=ON
+
```

Comment:
This build line is for everything, including for source that is not ours and that we build here. Such as the Azure PnP C SDK, or Parson. We do not want sanitizer to be run on that code and block a PR/fail CI for problems found there.

How are we going to avoid this kind of action?

Reply:
In my opinion, we should avoid such approach. If there is anything that's caught by sanitizers in 3rd party, it means the bug is reachable in our codebase and should be fixed before merging a particular PR. In addition to that, the first enablement of sanitizers should not trigger any bugs at current main, so I don't expect anything to show up within 3rd parties.

Reply:
We cannot easily fix code in the PnP C SDK, for example. The team that used to own that no longer exists. For us, that code is now disabled by default and only optionally enabled, however is still included in builds. Best is to see what happens, wanted to rase awareness of this however.